src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/update_turn_list_response.cc

```diff
@@ -57,6 +57,17 @@ UpdateTurnListResponse::~UpdateTurnListResponse() {}
 void UpdateTurnListResponse::Run() {
   LOG4CXX_AUTO_TRACE(logger_);
 
+  mobile_apis::Result::eType result_code =
+      static_cast<mobile_apis::Result::eType>(
+          (*message_)[strings::msg_params][strings::result_code].asInt());
+  ApplicationSharedPtr application = application_manager_.application(
+      (*message_)[strings::params][strings::connection_key].asInt());
+
+  if ((mobile_apis::Result::REJECTED != result_code) && application) {
+    application->UnsubscribeFromSoftButtons(
+        (*message_)[strings::params][strings::function_id].asInt());
+  }
```

Comment:
@IGapchuk , as a suggestion, to avoid duplication, we can make a method in app manager that accepts function id, connection key and result code, and performs unsubscribe if needed.

Reply:
@mked-luxoft Moved to app manager in 86f78e521e510b3a41cafe03b1b9bfed2c7b2433.